arbnode/batch_poster.go

```diff
@@ -208,6 +212,84 @@ func NewBatchPoster(l1Reader *headerreader.HeaderReader, inbox *InboxTracker, st
 	return b, nil
 }
 
+// checkRevert checks blocks with number in range [from, to] whether they
+// contain reverted batch_poster transaction.
+func (b *BatchPoster) checkReverts(ctx context.Context, from, to *big.Int) (bool, error) {
+	if from.Cmp(to) == 1 {
+		return false, fmt.Errorf("wrong range, from: %v is more to: %v", from.Int64(), to.Int64())
```

Comment:
This will get hit if `last ` is the same as `h.Number` (which can happen due to an L1 reorg), but I don't think we want to log a warning in that case.

Reply:
Changed the call site to set last as h.Number - 1 if it equals h.Number, since this function shouldn't be aware of outside logic, rather simply should check if there are any reverts in the specified block range.